code/game/objects/structures/shower.dm

```diff
@@ -252,3 +264,31 @@
 	layer = FLY_LAYER
 	anchored = TRUE
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/obj/item/showerparts
+	name = "shower parts"
+	desc = "All the things you need to construct a shower, comes with in-built H2O atom assembler to compensate for Nanotrasen's lack of plumbing installation."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "showerparts"
+	var/created_shower_type = /obj/machinery/shower
+
+/obj/item/showerparts/attack_self(mob/user)
+	. = ..()
+	user.visible_message("<span class='danger'>[user] starts putting together a shower.</span>", \
+	"<span class='userdanger'>You start putting together a shower!</span>")
+	if(do_after(user, 40))
+		if(QDELETED(src))
+			return FALSE
+		PlaceShower(user)
+
+/obj/item/showerparts/proc/PlaceShower(mob/user)
+	new created_shower_type(get_turf(user))
+	user.visible_message("<span class='danger'>[user] finishes putting together a shower.</span>", \
+	"<span class='userdanger'>You finish putting together the shower!</span>")
+	qdel(src)
```

Comment:
just put this in attack_self and save a proccall, my salary can only afford so much anon.

Reply:
okidoki